app-text/wordgrinder/wordgrinder-0.7.1.recipe

```diff
@@ -48,13 +50,38 @@ BUILD_PREREQUIRES="
 	cmd:pkg_config$secondaryArchSuffix
 	"
 
+if [ -n "$secondaryArchSuffix" ]; then
+SUMMARY_dictionaries="Built-in dictionaries for WordGrinder"
+ARCHITECTURES_dictionaries="any"
+PROVIDES_dictionaries="
+	wordgrinder_dictionaries = $portVersion
+	"
+REQUIRES_dictionaries="
+	wordgrinder == $portVersion base
+	"
+SUPPLEMENTS_dictionaries="
+	wordgrinder
+	"
+fi
+
 defineDebugInfoPackage wordgrinder$secondaryArchSuffix \
 	$commandBinDir/wordgrinder
 
+PATCH()
+{
+	sed -i -e "s|/etc/dictionaries-common/words|$dataDir/wordgrinder/dictionaries/|;" \
+		src/lua/addons/spillchocker.lua
+	sed -i -e "s|HOME ..||;" \
+		src/lua/settings.lua
+	sed -i -e "s|/.wordgrinder.settings|$USER_SETTINGS/wordgrinder/settings|;" \
+		src/lua/settings.lua
+}
+
 BUILD()
 {
 	make PREFIX=$prefix BINDIR=$commandBinDir DOCDIR=$docDir \
 		MANDIR=$manDir HOME="`finddir B_USER_DIRECTORY`" \
+		USER_SETTINGS_DIRECTORY=$USER_SETTINGS_DIRECTORY
```

Comment:
You don't need this line:
```
export USER_SETTINGS=`/bin/finddir B_USER_SETTINGS_DIRECTORY`
```
You just need to pass `$settingsDir/wordgrinder`
```diff
-		USER_SETTINGS_DIRECTORY=$USER_SETTINGS_DIRECTORY
+		USER_SETTINGS_DIRECTORY=$settingsDir/wordgrinder \
```

You will also need to insert this bloc after `SUPPLEMENTS_dictionaries` and before `PATCH()`:
```
USER_SETTINGS_FILES="
	settings/wordgrinder directory
```

Reply:
hm,I checked and I not found the USER_SETTINGS_DIRECTORY ,I check user settings stored in HOME,can I change the HOME to B_USER_SETTINGS_DIRECTIORY?

Reply:
I'm just try,for checks makefile have user settings var